src/lib/sidenav/sidenav.ts

```diff
@@ -161,7 +162,7 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
     let activeEl = this._doc && this._doc.activeElement;
     if (activeEl && this._elementRef.nativeElement.contains(activeEl)) {
       if (this._elementFocusedBeforeSidenavWasOpened instanceof HTMLElement) {
-        this._elementFocusedBeforeSidenavWasOpened.focus();
+        this._focusOriginMonitor.focusVia(this._elementFocusedBeforeSidenavWasOpened, 'program');
```

Comment:
I'm not sure how that would work,  considering that we don't have the equivalent of mdMenuTriggerFor for the sidenav. Alternatively, I can fix this only for the dialog where we have more control over the closing logic.